### PR TITLE
Component input validation

### PR DESCRIPTION
**Pull request by joelrebel**
This wraps up the remaining fixes and tests for the server components endpoints,

- Few more component field input validation and tests.
- Update component attributes in `componentUpdate` .

### CODE REVIEW

File: pkg/api/v1/router_server_components.go [hunk 1]
```diff
@@ -95,14 +102,30 @@ func (r *Router) serverComponentsCreate(c *gin.Context) {
 
 	// check server exists
 	if server == nil {
-		notFoundResponse(c, "no such server")
+		notFoundResponse(c, "server resource referenced by UUID does not exist: "+server.ID)
 		return
 	}
 
-	// components payload
+	//	// components payload
```
Comment (JAORMX) [inline pkg/api/v1/router_server_components.go:109]: can you keep the comment as it was?

File: pkg/api/v1/router_server.go [hunk 2]
```diff
@@ -100,6 +100,16 @@ func (r *Router) serverCreate(c *gin.Context) {
 		return
 	}
 
+	exists, err := models.ServerExists(c.Request.Context(), r.DB, srv.UUID.String())
```
Comment (JAORMX) [inline pkg/api/v1/router_server.go:103]: I thought the UUID was auto-generated.

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: pkg/api/v1/router_server.go (hunk 2 above)
Comment (joelrebel) [reply pkg/api/v1/router_server.go]: Nice find, I'll go test this separately again (now I wonder why it worked in my tests).

reverted commits as  e3cae69 23091ae

File: pkg/api/v1/router_server_components.go (hunk 1 above)
Comment (joelrebel) [reply pkg/api/v1/router_server_components.go]: fixed in [6a44696](https://github.com/metal-toolbox/hollow-serverservice/pull/116/commits/6a44696c84d0df7fde9f35d8a9e0edac1175ace1)

**Review by JAORMX — APPROVED** [review]

**Pull request merged**